### PR TITLE
[fixed] Strict mode error

### PR DESCRIPTION
**Pull request by grncdr**
Function declarations inside blocks is an error in strict mode. Certain versions of V8 actually notice this and throw an error.

/cc @elblivion

### CODE REVIEW

**Comment by grncdr** [comment]
Based on these two fixes, I think that tests are not running at all...

**Comment by elblivion** [comment]
Seems happier now, thanks! Now I'm getting this which is probably unrelated - don't know where this `tap` function is supposed to come from:

```
/Users/stanton/Code/contentful-management.js/example/clone-space.js:73
      .tap(logSummary)
       ^
TypeError: undefined is not a function
```

https://github.com/contentful/contentful-management.js/blob/master/example/clone-space.js#L73

**Comment by sdepold** [comment]
It would come from Bluebird which we aren't using anymore.

**Comment by elblivion** [comment]
In any case it's the `getSpace` doesn't seem to be working, I added some debug and it doesn't throw an error for a nonexistent space:

```
console.log('Getting source Space %s from %s using access token %s', sourceSpaceId, host, accessToken);
client.getSpace(sourceSpaceId)
      .catch(reportInvalidSpace);
//      .then(getDestinationSpace)
//      .tap(logSummary)
//      .spread(clone)
      //.done();
```
```
$ bash wrapper
Getting source Space obviouslyfake from api.contentful.com using access token <snip>
$ echo $?
0
```

File: example/clone-space.js [hunk 1]
```diff
@@ -193,6 +196,9 @@ function clone (sourceSpace, destinationSpace) {
   });
 }
 
+function tap$ (fn) {
```
Comment (sdepold) [inline example/clone-space.js]: not needed I guess?

Comment (grncdr) [inline example/clone-space.js]: Yeah that can be removed

File: lib/wait.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-var Promise = require('es6-promise');
+var Promise = require('es6-promise').Promise;
```
Comment (johanneswuerbach) [inline lib/wait.js]: Missing `use strict`.

**Comment by grncdr** [comment]
The last test run was an _actual_ test run. There are still some issues with the test running (builds may not fail correctly) so be careful when merging PRs in the future.

**Pull request merged**